### ISSUE TITLE
S2583: FP with multiple if-else clauses involving NaN

### ISSUE THREAD

**Issue by kamushkin** [SonarSource/SonarJS#618]
RSPEC-2583 - the following code triggers FP on the third condition -
```js
function compare1(a, b){
	return a < b ? -1 : a > b ? 1 : a >= b ? 0 : NaN;
}
```
https://sonarcloud.io/project/issues?id=kamushkin%3Asonarqube-tests&issues=AV0H2uTRI1Qj-tYTbHFX&open=AV0H2uTRI1Qj-tYTbHFX

Same when the code is written as if-else statements -
```js
function compare2(a, b){

	if (a < b){
		return -1;
	}
	else if (a > b){
		return 1;
	}
	else if (a >= b){
		return 0;
	}
	else {
		return NaN;
	}
}
```
https://sonarcloud.io/project/issues?id=kamushkin%3Asonarqube-tests&issues=AV0H2uTTI1Qj-tYTbHFY&open=AV0H2uTTI1Qj-tYTbHFY

original source -
https://sonarcloud.io/project/issues?id=d3&open=AVcREIlyzjiM7eGZNOsO&resolved=false&types=BUG

The FP is not triggered when the third comparison is `==`, so probably a bug.

**Comment by inverno** [comment]
@kamushkin thanks, yes, it's likely a bug

**Comment by Nisgrak** [comment]
This issue hasn't been resolved yet
Any case, I don't know if my error is relationed
```js
compare2(a) {
	a.forEach(b => {
		if (typeof b == "object") {
			return "aaa";
		} else if (Array.isArray(b)) { // This line is marked as error
			return "bbbb";
		} else {
			return 0;
		}
	});
}
```

**Comment by vilchik-elena** [comment]
This rule is deprecated as it's based on the Symbolic Execution engine we plan to drop soon. So updating the SonarLint plugin should disable the rule (but not in VSCode yet, but it should be the case soon).

So I closing this issue as won't fix, as rule will be dropped eventually.
